internal/pkg/manifest/svc.go

```diff
@@ -172,23 +196,52 @@ func (c *Count) Desired() (*int, error) {
 // Percentage represents a valid percentage integer ranging from 0 to 100.
 type Percentage int
 
+// Resource represents a resource that has autoscaling configurations.
+type Resource struct {
+	Value          *Percentage
+	ScaledResource AdvancedResource // mutually exclusive with Value
+}
+
+// AdvancedResource represents the allocation of a resource implementing scaling cooldown.
+type AdvancedResource struct {
+	Value    *Percentage `yaml:"value"`
+	Cooldown *Cooldown   `yaml:"cooldown"`
+}
+
+// Cooldown represents the autoscaling cooldown of resources.
+type Cooldown struct {
+	ScaleInCooldown  *time.Duration `yaml:"in"`
+	ScaleOutCooldown *time.Duration `yaml:"out"`
+}
+
 // AdvancedCount represents the configurable options for Auto Scaling as well as
 // Capacity configuration (spot).
 type AdvancedCount struct {
 	Spot         *int           `yaml:"spot"` // mutually exclusive with other fields
 	Range        Range          `yaml:"range"`
-	CPU          *Percentage    `yaml:"cpu_percentage"`
-	Memory       *Percentage    `yaml:"memory_percentage"`
+	Cooldown     *Cooldown      `yaml:"cooldown"`
```

Comment:
To be quite honest, I wrote as many things as pointers because I couldn't think of any reasons they wouldn't need to be, just for the memory advantage. I'll try to restructure them then to make it work with `mergo` better. 👍